### PR TITLE
Reorganize HDRP shader targets to share code between them (part 1)

### PR DESCRIPTION
**Pull request by alelievr**
# Purpose of this PR

⚠️  Please merge https://github.com/Unity-Technologies/Graphics/pull/604 before this one

HDRP sub-targets are now organized like this:
image

This part one refactors the way field, blocks and inspector settings of the HDRP targets are in the code to share the maximum of things between our different targets.

### CODE REVIEW

**Comment by alindmanUnity** [comment]
update: #604 has been merged to master stacks branch

File: com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.Migration.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        public bool TryUpgradeFromMasterNode(IMasterNode1 masterNode, out Dictionary<BlockFieldDescriptor, int> blockMap)
+        {
+            blockMap = null;
+            if(!(masterNode is DecalMasterNode1 decalMasterNode))
+                return false;
+            
+            // Set data
+            systemData.surfaceType = (SurfaceType)decalMasterNode.m_SurfaceType;
+            systemData.dotsInstancing = decalMasterNode.m_DOTSInstancing;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.Migration.cs:26]: We can remove dots instancing for decal, was only for V1 megacity

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.Migration.cs:26]: ok disregard, this was already there...

File: com.unity.render-pipelines.high-definition/Editor/Material/Eye/ShaderGraph/EyeSubrafceOptionPropertyBlock.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+using System;
```
Comment (elizabeth-legros) [inline com.unity.render-pipelines.high-definition/Editor/Material/Eye/ShaderGraph/EyeSubrafceOptionPropertyBlock.cs:1]: File name typo?

**Review by sebastienlagarde — APPROVED** [review]
Waiting for yamato fix - but overall code looks good
Shader error in 'BentNormals/BentNorm': invalid subscript 'BentNormal' at line 1007 (on d3d11)

**Review by elizabeth-legros — APPROVED** [review]
Other than file name typo looks good to me

File: com.unity.render-pipelines.high-definition/Editor/Material/Eye/ShaderGraph/EyeSubrafceOptionPropertyBlock.cs (hunk 2 above)
Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Material/Eye/ShaderGraph/EyeSubrafceOptionPropertyBlock.cs]: fixed, thanks

**Pull request merged**